quill-sql/src/main/scala/io/getquill/sources/sql/SqlQuery.scala

```diff
@@ -167,6 +168,10 @@ object SqlQuery {
             select = select(alias)
```

Comment:
I think we still need to add a condition to the line 106: `if (!b.distinct && agg.isEmpty)`. The sql generation probably will break for non-isomorphic maps:

`query[Test].map(_.i).distinct.map(_ + 1)`

This kind of query will require nesting.